src/fu-engine.c

```diff
@@ -229,8 +229,14 @@ fu_engine_get_release_version (FuEngine *self, XbNode *component, XbNode *rel)
 	version_format = xb_node_query_text (component,
 					     "custom/value[@key='LVFS::VersionFormat']",
 					     NULL);
-	if (version_format != NULL)
+	if (version_format != NULL) {
 		fmt = fu_common_version_format_from_string (version_format);
+		if (fmt == FU_VERSION_FORMAT_UNKNOWN) {
+			g_warning ("version format %s unsupported, assuming plain",
+				   version_format);
+			return g_strdup (version);
```

Comment:
This is a behavioral change isn't it?  If the format string can't be determined it previously would default to triplet, but now it defaults to plain.

Reply:
I don't think there is a good default we can choose here. Maybe we should be more aggressive and return an error in this case?

Reply:
Yes, I think that makes sense.

Reply:
Agreed; it's made the diff a bit bigger but I think it's the right thing to do.